compiler/src/org.graalvm.compiler.core.amd64/src/org/graalvm/compiler/core/amd64/AMD64ArithmeticLIRGenerator.java

```diff
@@ -1162,6 +1335,23 @@ public Variable emitLoad(LIRKind kind, Value address, LIRFrameState state) {
         return result;
     }
 
+    @Override
+    public Variable emitVectorLoad(LIRKind vectorKind, int count, Value address, LIRFrameState state) {
+        AMD64AddressValue loadAddress = getAMD64LIRGen().asAddressValue(address);
+        Variable result = getLIRGen().newVariable(vectorKind);
+        // getLIRGen().append(new AMD64Unary.VectorReadMemory(result, loadAddress));
+        // Use the line below instead of the line above for the temporary-stack-space load op.
+        getLIRGen().append(new AMD64Packing.LoadStackOp(getLIRGen(), asAllocatable(result), loadAddress, count));
```

Comment:
Currently we go via the stack to support arbitrary-size vectors. These could be slow and there might be a better approach.